bql/semantic/hooks.go

```diff
@@ -664,6 +671,59 @@ func whereObjectClause() ElementHook {
 	return f
 }
 
+// whereFilterClause returns an element hook that updates the working filter clause and,
+// if the filter clause is complete, populates the filters list of the statement.
+func whereFilterClause() ElementHook {
+	var hook ElementHook
+	supportedFilterFunctions := map[string]bool{
+		"latest": true,
+	}
+
+	hook = func(st *Statement, ce ConsumedElement) (ElementHook, error) {
```

Comment:
This seems like a state machine,
 the state is given by st and ce and it modifies it and gets called again with the modified state until completion.

Would you mind explaining that?

Reply:
I believe the comment 3 lines above for `whereFilterClause` is already explaining this:

```
// whereFilterClause returns an element hook that updates the working filter clause and,
// if the filter clause is complete, populates the filters list of the statement.
```

But, if you believe we still need to add any other comments just lmk and I will be glad to do so.